--- trunk/gcc/cppspec.c	2010/08/22 12:19:01	163459
+++ trunk/gcc/cppspec.c	2011/05/16 18:34:31	173801
@@ -1,5 +1,5 @@
 /* Specific flags and argument handling of the C preprocessor.
-   Copyright (C) 1999, 2007, 2010 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2007, 2010, 2011 Free Software Foundation, Inc.
 
 This file is part of GCC.
 
@@ -158,7 +158,7 @@
 
   new_decoded_options = XNEWVEC (struct cl_decoded_option, new_argc);
 
-  new_decoded_options[0] = new_decoded_options[0];
+  new_decoded_options[0] = decoded_options[0];
   j = 1;
 
   if (need_E)
